crates/ra_hir_ty/src/lib.rs

```diff
@@ -680,6 +680,21 @@ impl Ty {
         }
     }
 
+    pub fn deref(&self) -> &Ty {
+        let mut t: &Ty = self;
+
+        loop {
+            match t {
```

Comment:
this could probably be a `while let`

Reply:
`while let` cleaned this up a bit, thanks for the suggestion.